source/common/network/connection_impl.cc

```diff
@@ -39,6 +39,70 @@ void ConnectionImplUtility::updateBufferStats(uint64_t delta, uint64_t new_total
   }
 }
 
+void DynamicSocketOptionsImpl::setSocketRecvBufferSize(uint32_t buffer_size) const {
+  int rc = 0;
+  if (io_handle_.fd() == -1) {
+    ENVOY_CONN_LOG(trace, "Stale socket file handle, operation Set Receive Buffer aborted", *this);
+    return;
+  }
+  // Modify the receive buffer size to 'buffer_size'
+  rc = setsockopt(io_handle_.fd(), SOL_SOCKET, SO_RCVBUF, &buffer_size, sizeof(uint32_t));
```

Comment:
OsSysCallsImpl::setsockopt?
ditto getsockopt

Reply:
Sure. Did not notice it earlier. It is a wrapper on this API. I will make this change.